### PR TITLE
DL-4353 Check on IABD 55 amounts

### PR DESCRIPTION
**Pull request by grahampaulcook**
# DL-4353 Check on IABD 55 amounts

**New feature**

Additional check added for IABD 55's

### CODE REVIEW

File: app/models/OtherExpense.scala [hunk 1]
```diff
@@ -18,8 +18,8 @@ package models
```
Comment (alijy) [inline app/models/OtherExpense.scala:18]: Better to name this file `IABDExpense.scala`

Comment (alijy) [inline app/models/OtherExpense.scala:18]: Better to name this file `IABDExpense.scala`

File: app/connectors/TaiConnector.scala [hunk 2]
```diff
@@ -18,40 +18,29 @@ package connectors
...
-        }
-      case _ =>
-        Seq.empty
-    }
+  def getOtherExpensesData(nino: String, year: Int)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[IABDExpense]] = {
+    getIabdData(nino, year, appConfig.otherExpensesId)
+  }
+
+  def getJobExpensesData(nino: String, year: Int)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[IABDExpense]] = {
+    getIabdData(nino, year, appConfig.jobExpenseId)
```
Comment (alijy) [inline app/connectors/TaiConnector.scala:38]: These 2 functions are only called once from `CheckAlreadyClaimedAction`, so why not just calling `getIabdData` directly and removing these 2 functions?

Comment (alijy) [reply app/connectors/TaiConnector.scala:38]: That way, if there are other IABD codes we need to check, there will be no need to create more functions to call `getIabdData`.

File: app/models/OtherExpense.scala (hunk 1 above)
Comment (grahampaulcook) [reply app/models/OtherExpense.scala:18]: Pushed a commit with a rename of the file.

File: app/connectors/TaiConnector.scala (hunk 2 above)
Comment (grahampaulcook) [reply app/connectors/TaiConnector.scala:38]: I'm fairly confident that we would not need to add further IABD codes to check. Your suggestion would also work perfectly fine but I suppose the way I've done it is to lock down what can be asked for. You are right that we do not know if they will ask for further codes but equally they could ask for IABD 55's to pass a different HTTP header compared to IABD 59's or TAI could change the endpoint for a IABD 55 compared to an IABD 59. If that was to occur then we would end up with the single `getIabdData` with if statements in it.

Comment (alijy) [reply app/connectors/TaiConnector.scala:38]: I agree

**Review by alijy — APPROVED** [review]

**Pull request merged**